pkg/apis/core/validation/shoot.go

```diff
@@ -1101,6 +1097,12 @@ func ValidateWorkers(workers []core.Worker, fldPath *field.Path) field.ErrorList
 				atLeastOnePoolWithCompatibleTaints = true
 			}
 		}
+
+		if atLeastOneActivePool && atLeastOnePoolWithCompatibleTaints {
+			if worker.SystemComponents != nil && worker.SystemComponents.Allow != nil && *worker.SystemComponents.Allow {
+				atLeastOnePoolWithAllowedSystemComponents = true
+			}
+		}
```

Comment:
Hmmm, I think this will allow Shoots with a worker section like this, although we will not be able to schedule system components on either worker pools:
```
workers:
- name: untainted-but-without-system-components
  systemComponents:
    allow: false
- name: tainted-but-system-components-allowed
  taints:
  - key: Foo
     effect: NoSchedule
```
Correct?
I think, we should check here, if the **current** `worker` has no or only compatible taints `systemComponents.allowed=true`.

Reply:
Done